packages/cozy-harvest-lib/src/components/TriggerManager.jsx

```diff
@@ -27,41 +27,112 @@ import TriggerLauncher from './TriggerLauncher'
 import VaultCiphersList from './VaultCiphersList'
 import manifest from '../helpers/manifest'
 import HarvestVaultProvider from './HarvestVaultProvider'
+import clone from 'lodash/clone'
+import flag from 'cozy-flags'
 
 import { createOrUpdateCipher } from '../models/cipherUtils'
+import { konnectorPolicy as biKonnectorPolicy } from '../services/budget-insight'
+
+const defaultKonnectorPolicy = {
+  accountContainsAuth: true,
+  saveInVault: true,
+  onAccountCreation: null,
+  match: () => true
+}
+
+const policies = [
+  flag('bi-konnector-policy') ? biKonnectorPolicy : null,
+  defaultKonnectorPolicy
+].filter(Boolean)
 
 const IDLE = 'IDLE'
 const RUNNING = 'RUNNING'
 
 const MODAL_PLACE_ID = 'coz-harvest-modal-place'
 
-const buildOrUpdateAccount = ({ account, userData, cipher, konnector }) => {
+const findKonnectorPolicy = konnector =>
+  policies.find(policy => policy.match(konnector))
+
+/**
+ * Creates or updates an io.cozy.accounts
+ * Used as a form submit handler
+ *
+ * @param  {io.cozy.account} options.account - Existing io.cozy.account or object
+ * @param  {[type]} options.cipher - Vault cipher if vault has been unlocked
+ * @param  {CozyClient} options.client - A cozy client
+ * @param  {io.cozy.konnector} options.konnector - Konnector to which the account is linked
+ * @param  {KonnectorPolicy} options.konnectorPolicy - Controls if auth is saved in io.cozy.accounts
+ * and if auth is saved into the vault
+ * @param  {function} options.saveAccount
+ * @param  {object} options.userData
+ * @param  {function} options.ensureTrigger
+ */
+const createOrUpdateAccount = async ({
+  account,
+  cipher,
+  client,
+  konnector,
+  konnectorPolicy,
+  saveAccount,
+  userCredentials
+}) => {
   const isUpdate = !!account
+  const { onAccountCreation, saveInVault } = konnectorPolicy
+
+  let accountToSave = clone(account)
 
-  let accountToSave
+  accountToSave = accounts.resetState(accountToSave)
 
   accountToSave = accounts.setSessionResetIfNecessary(
-    accounts.resetState(account),
-    userData
+    accountToSave,
+    userCredentials
   )
 
   accountToSave = isUpdate
-    ? accounts.mergeAuth(accountToSave, userData)
-    : accounts.build(konnector, userData)
+    ? accounts.mergeAuth(accountToSave, userCredentials)
+    : accounts.build(konnector, userCredentials)
 
-  if (cipher) {
+  if (onAccountCreation) {
+    accountToSave = await onAccountCreation({
+      client,
+      account: accountToSave,
+      konnector,
+      saveAccount
```

Comment:
I'm not sure I understand why we pass `saveAccount`, it's not really the policie's job to actually save the account?

Reply:
For the budget-insight policy, we have to save the account prior to launching the temporary token job, since we have to have an account to launch a konnector job.